tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -238,6 +247,43 @@ function get_tdx_ovmf_path_from_distro() {
 	esac
 }
 
+function adjust_qemu_cmdline() {
+	shim="${1}"
+	config_path="${2}"
+	qemu_share="${shim}"
+
+	# The paths on the kata-containers tarball side look like:
+	# ${dest_dir}/opt/kata/share/kata-qemu/qemu
+	# ${dest_dir}/opt/kata/share/kata-qemu-snp-experimnental/qemu
+	[[ "${shim}" =~ ^(qemu-snp|qemu-nvidia-snp)$ ]] && qemu_share=${shim}-experimental
+		
+	qemu_binary=$(tomlq '.hypervisor.qemu.path' ${config_path} | tr -d \")
+	qemu_binary_script="${qemu_binary}-installation-prefix"
+	qemu_binary_script_host_path="/host/${qemu_binary_script}"
+
+	if [[ ! -f ${qemu_binary_script_host_path} ]]; then
+		# From the QEMU man page:
+		# ```
+		# -L  path
+		# 	Set the directory for the BIOS, VGA BIOS and keymaps.
+		# 	To list all the data directories, use -L help.
+		# ```
+		#
+		# The reason we have to do this here, is because otherwise QEMU
+		# will only look for those files in specific paths, which are
+		# tied to the location of the PREFIX used during build time
+		# (/opt/kata, in our case).
+		cat <<EOF >${qemu_binary_script_host_path}
+#!/usr/bin/env bash
+
+${qemu_binary} "\$@" -L ${dest_dir}/share/kata-${qemu_share}/qemu/
```

Comment:
I strongly suspect `$@` shouldn't be escaped.

I'd also `exec` this for the sake of not keeping the `bash` process around for no reason.

Reply:
> I strongly suspect `$@` shouldn't be escaped.
It does work as expected as it is, but I can give it a try without it as well.

> I'd also exec this for the sake of not keeping the bash process around for no reason.
Makes sense.

Reply:
Sorry, for some reason I read "escaped" but my mind processed as "quoted".

We're not escaping `$@`, but we're ensuring that `$` will be in the final script rather than treating `$@` as an environment variable during the `cat` time.

Reply:
You're right :smile: